src/id-data-controls.tsx

```diff
@@ -0,0 +1,271 @@
+import { render } from 'preact';
+import { SavedIdData } from './DataType/SavedIdData';
+import { Privacy, PRIVACY_ACTIONS } from './Utility/Privacy';
+import t from './Utility/i18n';
+import { IntlProvider, MarkupText, Text } from 'preact-i18n';
+import { DynamicInputContainer } from './Components/Generator/DynamicInputContainer';
+import { AddressControl, DateControl, InputControl } from './Components/Generator/DynamicInput';
+import { SignatureInput } from './Components/Generator/SignatureInput';
+import { FeatureDisabledWidget } from './Components/FeatureDisabledWidget';
+import { useCallback, useEffect, useMemo, useState } from 'preact/hooks';
+import { Address, IdDataElement, Signature } from 'request';
+import { produce } from 'immer';
+import { isAddress } from './Utility/requests';
+
+const DEFAULT_FIXED_FIELDS = {
+    name: '',
+    birthdate: '',
+    email: '',
+    address: {
+        street_1: '',
+        street_2: '',
+        place: '',
+        country: '',
+        primary: true,
+    },
+};
+
+const IdDataControls = () => {
+    const savedIdData = useMemo(() => new SavedIdData(), []);
+    const [custom_id_data, setCustomIdData] = useState<IdDataElement[]>([]);
+    const [signature, setSignature] = useState<Signature>({ type: 'text', name: '' });
+    const [fixed_id_data, setFixedIdData] = useState(DEFAULT_FIXED_FIELDS);
+
+    const resetSavedIdData = useCallback(() => {
+        savedIdData.getAll().then((id_data) => id_data && setCustomIdData(id_data));
+        savedIdData.getAllFixed().then((fixed_data) => {
+            setFixedIdData(
+                (fixed_data ?? []).reduce(
+                    (acc, elem: IdDataElement) => ({ ...acc, [elem.type]: elem.value }),
+                    DEFAULT_FIXED_FIELDS
+                )
+            );
+        });
+        savedIdData.getSignature().then((signature) => signature && setSignature(signature));
+    }, [savedIdData]);
+
+    useEffect(() => {
+        resetSavedIdData();
+    }, [resetSavedIdData]);
+
+    useEffect(() => {
+        savedIdData.storeArray(fieldsArrayFromFixedData(fixed_id_data));
+    }, [savedIdData, fixed_id_data]);
+
+    useEffect(() => {
+        savedIdData.storeArray(custom_id_data, false);
+    }, [savedIdData, custom_id_data]);
```

Comment:
Don't these cause an unnecessary write on load?

Reply:
Yes, but they do not affect the state. So I don't care until a better solution comes along.